application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -0,0 +1,182 @@
+
+        TextInput transferQuestionTag = TextInput
+            .create(TRANSFER_QUESTION_TAG, "Transfer question tags", TextInputStyle.SHORT)
+            .setValue("Java")
+            .build();
+
+        String transferQuestionModalComponentID =
+                generateComponentId(authorId, originalMessageId, originalChannelId);
+        Modal transferModal =
+                Modal.create(transferQuestionModalComponentID, "transfer question menu")
+                    .addActionRow(transferQuestionTitle)
+                    .addActionRow(transferQuestionInput)
+                    .addActionRow(transferQuestionTag)
+                    .build();
+
+        event.replyModal(transferModal)
+            .queue(success -> logger.info(
+                    "{} with id: {}  triggered the transfer action on message with id: {}",
+                    event.getUser().getName(), event.getUser().getId(), originalMessageId),
+                    failed -> {
+                    });
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        event.getJDA()
+            .retrieveUserById(args.get(0))
+            .map(user -> user)
+            .flatMap(fetchedUser -> createForumPost(event, fetchedUser))
+            .flatMap(user -> dmUser(event.getChannel(), user))
+            .flatMap(helperForum -> event.reply("transferred to" + helperForum.getAsMention())
+                .setEphemeral(true))
+            .flatMap(info -> deleteOriginalMessage(event.getJDA(), args.get(2), args.get(1)))
+            .queue();
+    }
+
+    private static String createTitle(String message) {
+        if (message.length() >= TITLE_MAX_LENGTH) {
+            int lastWordEnd = message.lastIndexOf(' ', TITLE_MAX_LENGTH);
+
+            if (lastWordEnd == -1) {
+                lastWordEnd = TITLE_MAX_LENGTH;
+            }
+
+            message = message.substring(0, lastWordEnd);
+        }
+
+        return isTitleValid(message) ? message : "Untitled";
+    }
+
+    private static boolean isTitleValid(CharSequence title) {
+        String titleCompact = TITLE_COMPACT_REMOVAL_PATTERN.matcher(title).replaceAll("");
+
+        return titleCompact.length() >= TITLE_COMPACT_LENGTH_MIN
+                && titleCompact.length() <= TITLE_COMPACT_LENGTH_MAX;
+    }
+
+    private RestAction<User> createForumPost(ModalInteractionEvent event, User originalUser) {
+        MessageCreateData forumMessage = MessageCreateData
+            .fromContent(event.getValue(TRANSFER_QUESTION_INPUT_ID).getAsString());
+        String forumTitle = event.getValue(TRANSFER_QUESTION_TITLE_ID).getAsString();
+        ForumChannel questionsForum = getHelperForum(event.getJDA());
+        List<ForumTag> tags = questionsForum.getAvailableTagsByName("JAVA", true);
+        ForumTag javaTag = tags.get(0);
+
+        return questionsForum.createForumPost(forumTitle, forumMessage)
+            .setTags(ForumTagSnowflake.fromId(javaTag.getId()))
```

Comment:
if not select menu, i'll suggest to make the post without any tags (if possible)

Reply:
Questions forum madates the need for a  tag when creating new ones, im not sure if i can bypass that without a tag.

Reply:
so, what other options do u have? i can think of putting all the tags by default separated by a comma and then the user can delete the unnecessary ones, what do u think?

Reply:
is it possible to do select menu in modal? i don't like putting all tags in, gonna clutter the field. We could also rely on `/help-thread-change-category`.

Reply:
> is it possible to do select menu in modal?

no, not possible

> i don't like putting all tags in, gonna clutter the field. We could also rely on /help-thread-change-category

yes, but we'll have to put some tag in the beginning as u said its not possible to create the thread without a tag

Reply:
`Java` tag is put by default, altho it's impossible that this will be gone in future(being a java oriented server) but i can also pull first tag from config.

Reply:
> altho it's impossible that this will be gone in future(being a java oriented server)

this bot is designed to be deployed for any server

> i can also pull first tag from config

i'd suggest to add the last tag, coz that'll add the least number of members (probably), and then it can be customized after creation

Reply:
> this bot is designed to be deployed for any server

Interesting, do we support any server atm?
 
> i'd suggest to add the last tag, coz that'll add the least number of members (probably), and then it can be customized after creation

ill start looking into that

Reply:
> Interesting, do we support any server atm?

idk, maybe someone somewhere is hosting this bot for their server as well, but that is very unlikely. We don't give any special support if someone wants to, if that's what ur asking